modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperienceLocalServiceImpl.java

```diff
@@ -236,6 +240,50 @@ public void deleteSegmentsExperiences(
 		}
 	}
 
+	@Override
+	public SegmentsExperience duplicateSegmentsExperience(
+			Locale locale, long segmentsExperienceId,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		SegmentsExperience segmentsExperience =
+			segmentsExperiencePersistence.findByPrimaryKey(
+				segmentsExperienceId);
+
+		List<SegmentsExperience> segmentsExperiences = new ArrayList<>(
+			segmentsExperiencePersistence.findByG_C_C_GEP(
+				segmentsExperience.getGroupId(),
+				segmentsExperience.getClassNameId(),
+				segmentsExperience.getClassPK(),
+				segmentsExperience.getPriority()));
+
+		for (SegmentsExperience curSegmentsExperience : segmentsExperiences) {
```

Comment:
We should call the method updateSegmentsExperiencePriority, that contains the logic to update the priority of a segment. 

I think would be better to:

1. add the experience, in the default position.
2. update the priority calling updateSegmentsExperiencePriority (maybe we can change the swap method (contemplate n swaps until you find your position that is what you are doing in this for, or contemplate this new option as a new method).

Reply:
I see your point but don't you think that the implementation provided is much efficiency? Actually the one you just proposed was my first approach. Then I realized that the deleteExperince method is doing something similar

Reply:
apart from that, when you said in the step one 

> add the experience, in the default position.

You meant the position that the service provides for the new experience by default (the highest) and not the priority of the default experience, right?